src/app/components/BibPage/BibDetails.jsx

```diff
@@ -174,11 +174,49 @@ class BibDetails extends React.Component {
     );
   }
 
+  constructSubjectHeading(bibValue, url, fieldValue, fieldLabel) {
+    let currentArrayString = '';
+    const singleSubjectHeadingArray = bibValue.split(' > ');
+    const returnArray = [];
+
+    const urlArray = url.split(' > ').map((urlString, index) => {
+      const dashDivided = (index !== 0) ? ' -- ' : '';
+      currentArrayString = `${currentArrayString}${dashDivided}${urlString}`;
+
+      return currentArrayString;
+    });
+
+    const linkArray = singleSubjectHeadingArray.map((heading, index) => {
+      return(
+        <Link
+          onClick={e => this.newSearch(e, url, fieldValue, bibValue, fieldLabel)}
+          to={`${appConfig.baseUrl}/search?${urlArray[index]}.`}
+          key={index}
+        >
+          {heading}
+        </Link>
+      );
+    });
+
+    linkArray.forEach((linkElement, index) => {
```

Comment:
It seems like this can be merged with the function above, so you don't have to do two loops, and conditionally add the `span`.

Reply:
Good idea!